scripts/globals/mobskills/cosmic_elucidation.lua

```diff
@@ -0,0 +1,35 @@
+---------------------------------------------------
+--  Cosmic Elucidation
+--  Description: Cosmic Elucidation inflicts heavy AOE damage to everyone in the battle.
+--  Type:
+--  Utsusemi/Blink absorb: Ignores shadows
+--  Range:
+--  Notes: Ejects all combatants from the battlefield, resulting in a failure.
+---------------------------------------------------
+require("settings/main")
+require("scripts/globals/status")
+require("scripts/globals/mobskills")
+require("scripts/globals/msg")
+---------------------------------------------------
+local mobskill_object = {}
+
+mobskill_object.onMobSkillCheck = function(target, mob, skill)
+    return 1 -- only scripted use
+end
+
+mobskill_object.onMobWeaponSkill = function(target, mob, skill)
+    local dmgmod = 2
+local info = xi.mobskills.mobMagicalMove(mob, target, skill, mob:getWeaponDmg()*21, xi.magic.ele.LIGHT, dmgmod, 3)
```

Comment:
a few things:

1) indentation needs fixed
2) you are missing the final param of mobMagicalMove "tpValue", which is nil here. You have set the tpeffect to "3", see function header: `xi.mobskills.mobMagicalMove = function(mob, target, skill, damage, element, dmgmod, tpeffect, tpvalue)
`
3) please update the modifier to the enum value and not just a number (see below)
4) assign the tpValue final parameter to something sane (from the mobskill.lua):

```
-- MAGICAL MOVE
-- Call this on every magical move!
-- mob/target/skill should be passed from onMobWeaponSkill.
-- dmg is the base damage (V value), accmod is a multiplier for accuracy (1 default, more than 1 = higher macc for mob),
-- ditto for dmg mod but more damage >1 (equivalent of M value)
-- tpeffect is an enum from one of:
-- 0 = xi.mobskills.magicalTpBonus.NO_EFFECT
-- 1 = xi.mobskills.magicalTpBonus.MACC_BONUS
-- 2 = xi.mobskills.magicalTpBonus.MAB_BONUS
-- 3 = xi.mobskills.magicalTpBonus.DMG_BONUS
-- tpvalue affects the strength of having more TP along the following lines:
-- xi.mobskills.magicalTpBonus.NO_EFFECT -> tpvalue has no xi.effect.
-- xi.mobskills.magicalTpBonus.MACC_BONUS -> direct multiplier to macc (1 for default)
-- xi.mobskills.magicalTpBonus.MAB_BONUS -> direct multiplier to mab (1 for default)
-- xi.mobskills.magicalTpBonus.DMG_BONUS -> direct multiplier to damage (V+dINT) (1 for default)
--Examples:
-- xi.mobskills.magicalTpBonus.DMG_BONUS and TP=100, tpvalue = 1, assume V=150  --> damage is now 150*(TP*1)/100 = 150
-- xi.mobskills.magicalTpBonus.DMG_BONUS and TP=200, tpvalue = 1, assume V=150  --> damage is now 150*(TP*1)/100 = 300
-- xi.mobskills.magicalTpBonus.DMG_BONUS and TP=100, tpvalue = 2, assume V=150  --> damage is now 150*(TP*2)/100 = 300
-- xi.mobskills.magicalTpBonus.DMG_BONUS and TP=200, tpvalue = 2, assume V=150  --> damage is now 150*(TP*2)/100 = 600
```
4) you can now uncomment skill:setMsg(302) because the script was returning early and causing the return value to be nil, see pic related (I hardcoded it to 666)
![image](https://user-images.githubusercontent.com/60417494/177224732-03e1a199-388c-4fbd-aa73-705f63cf1c39.png)

5) I am sure nobody is sure if this is "light" elemental or not. I would probably assign it to be both `xi.attackType.SPECIAL` and `xi.damageType.ELEMENTAL`.  I don't think it matters because all this move is does is be pretty and the real damage is mostly irrelevant. Special and elemental should bypass most absorb checks while still using stoneskin/phalanx as intended.
6) There is no healed message for Cosmic Elucidation, so set `dmg = math.min(0, dmg)` after the `xi.mobskills.mobFinalAdjustments` call so the message doesn't return some huge bugged number.

Reply:
also, make sure to test the whole fight again and check for lua errors in xi_map's output

I only found out about the missing parameter because I wanted to see why the message didn't work